images/builder/debug-wrapper.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	bin, err := os.Executable()
+	if err != nil {
+		log.Println(err)
+		os.Exit(1)
+	}
+
+	inputArts := os.Args[1:]
+	debugPort, ok := os.LookupEnv("DEBUG_PORT")
+	if !ok || debugPort == "" {
+		debugPort = "2345"
+	}
+
+	debugContinue := true
+	if debugHold, ok := os.LookupEnv("DEBUG_HOLD"); ok && debugHold == "true" {
+		debugContinue = false
+	}
+
+	args := []string{
+		fmt.Sprintf("--listen=:%s", debugPort),
+		"--headless=true",
+		fmt.Sprintf("--continue=%t", debugContinue),
+		"--log=true",
+		"--log-output=debugger,debuglineerr,gdbwire,lldbout,rpc",
+		"--accept-multiclient",
+		"--api-version=2",
+		"exec",
+		fmt.Sprintf("%s-bin", bin),
+		"--",
+	}
+
+	args = append(args, inputArts...)
+
+	cmd := exec.Command("/usr/bin/dlv", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err = cmd.Run(); err != nil {
+		log.Printf("failed to execute dlv: %v", err)
```

Comment:
Do we need to filter the error types here, such as for `*exec.ExitError`? The error that is printed to the stdout here may be a bit confusing if everything is successful except that cilium-agent returns with exit code 1 for instance.

Reply:
good point. refactored the part that `ExitErrors` are handled separately by suppressing a log message and returning the exit code.

all other errors are handled by logging it and returning with exit code 1